src/addon/NUOPC/src/NUOPC_Driver.F90

```diff
@@ -994,113 +994,120 @@ recursive subroutine InitializeIPDv02p1(driver, importState, exportState, &
-              ! automatically created connectors inherit Profiling from parent
-              call NUOPC_CompAttributeGet(driver, name="Profiling", &
-                value=valueString, rc=rc)
-              if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, &
-                line=__LINE__, file=trim(name)//":"//FILENAME)) return  ! bail
-              if (trim(valueString)=="max" .or. trim(valueString)=="high" .or. &
-                trim(valueString)=="low" .or. trim(valueString)=="off") then
-                ! directly inherit presets
-                vString = trim(valueString)
+                  line=__LINE__, file=trim(name)//":"//FILENAME)) return
               else
-                ! not a preset level: lower 16-bit of parent's profiling setting
-                vInherit = ibits(profiling,0,16)
-                write(vString,"(I10)") vInherit
+                ! j==0
+                needConnector = needConnector.and. &
+                  ESMF_StateIsCreated(is%wrap%modelES(j), rc=rc)
+                if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, &
+                  line=__LINE__, file=trim(name)//":"//FILENAME)) return
               endif
-              if (btest(verbosity,13)) then
-                write (msgString,"(A)") trim(name)//&
-                  " - Setting Profiling on created component to: "// &
-                  trim(vString)
-                call ESMF_LogWrite(msgString, ESMF_LOGMSG_INFO, rc=rc)
+              if (needConnector) then
+                ! driver import or export States exist and connection requested
+                ! -> automatic connector add
+                call NUOPC_DriverAddComp(driver, &
+                  srcCompLabel=srcCompLabel, dstCompLabel=dstCompLabel, &
+                  compSetServicesRoutine=cplSS, rc=rc)
                 if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, &
-                  line=__LINE__, &
-                  file=trim(name)//":"//FILENAME, rcToReturn=rc)) &
-                  return  ! bail out
+                  line=__LINE__, file=trim(name)//":"//FILENAME)) return  ! bail
+                ! retrieve the component with petList
+                call NUOPC_DriverGetComp(driver, srcCompLabel, dstCompLabel, &
+                  comp=connector, petList=petList, relaxedflag=.true., rc=rc)
+                if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, &
+                  line=__LINE__, file=trim(name)//":"//FILENAME)) return  ! bail
+                ! automatically created connectors inherit Verbosity from parent
+                call NUOPC_CompAttributeGet(driver, name="Verbosity", &
+                  value=valueString, rc=rc)
+                if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, &
+                  line=__LINE__, file=trim(name)//":"//FILENAME)) return  ! bail
+                if (trim(valueString)=="max" .or. trim(valueString)=="high" .or. &
```

Comment:
Why not all non-blank values?

Reply:
Automatically created connectors inherit from the parent component (driver) that creates them. The meaning of Verbosity bits for Driver and Connector  agree for the lower 8 bits, therefore only inherit those, if this is a bit pattern. for presets, they are interpreted according to the NUOPC component kind, so presets can just be inherited directly. Does that make sense?

Reply:
It would make more sense if the connector contained metadata saying it was automatically created and if it was automatically created then it always inherits the lower 8 bits, instead of checking the value for "max","high", etc

Reply:
"max", "high", "low", and "off" are inherited verbatim. Any of these might define higher bits than the lower 8. However, the definition is component kind dependent, and thus must be done verbatim. If not a string, but an explicit bit pattern is found in the parent Verbosity setting, then the handling for inheritance is done as bit pattern, and only lower 8 bits are inherited. So the inheritance handling depends on the parents' Verbosity format. Same applies to Profiling.

Reply:
Well, that's not exactly what I meant but okay. The setup is a little confusing and therefore it will be difficult to maintain in the future.